raiden_contracts/tests/test_channel_unlock.py

```diff
@@ -64,6 +66,141 @@ def test_merkle_root(
     assert unlocked_amount == 9
 
 
+def test_unlock_wrong_locksroot(
+        web3,
+        token_network,
+        create_settled_channel,
+        get_accounts
+):
+    (A, B) = get_accounts(2)
+    settle_timeout = 8
+
+    pending_transfers_tree_A = get_pending_transfers_tree(web3, [1, 3, 5], [], settle_timeout)
+    pending_transfers_tree_A_fake = get_pending_transfers_tree(web3, [1, 3, 6], [], settle_timeout)
+
+    create_settled_channel(
+        A,
+        pending_transfers_tree_A.locked_amount,
+        pending_transfers_tree_A.merkle_root,
+        B,
+        0,
+        fake_bytes(32),
+        settle_timeout
+    )
+
+    with pytest.raises(TransactionFailed):
+        token_network.transact().unlock(
+            A,
+            B,
+            pending_transfers_tree_A_fake.packed_transfers
+        )
+
+
+def test_channel_unlock_bigger_locked_amount(
+        web3,
+        token_network,
+        create_settled_channel,
+        get_accounts,
+        reveal_secrets
+):
+    (A, B) = get_accounts(2)
+    settle_timeout = 8
+
+    # Mock pending transfers data
+    pending_transfers_tree_A = get_pending_transfers_tree(web3, [1, 3, 5], [2, 4], settle_timeout)
+    reveal_secrets(A, pending_transfers_tree_A.unlockable)
+
+    create_settled_channel(
+        A,
+        pending_transfers_tree_A.locked_amount + 1,
+        pending_transfers_tree_A.merkle_root,
+        B,
+        0,
+        fake_bytes(32),
+        settle_timeout
+    )
+
+    # This should pass, even though the locked amount in storage is bigger.
+    # But those locked tokens will remain locked in the contract
+    token_network.transact().unlock(
```

Comment:
perhaps here (and in the next test) also add a check for the token balance of the contract to check they are still locked there?